app/frontend/components/pages/login/mnemonicAuth.tsx

```diff
@@ -1,149 +1,137 @@
     loadWallet,
-    shouldShowMnemonicInfoAlert,
     openGenerateMnemonicDialog,
-  }) {
-    const sanitizedMnemonic = sanitizeMnemonic(formData.mnemonicInputValue)
+  } = useActions(actions)
 
-    return (
-      <div className={`authentication-content ${shouldShowMnemonicInfoAlert ? '' : 'centered'}`}>
-        {shouldShowMnemonicInfoAlert && (
-          <Alert alertType="info auth">
-            Here you can use your mnemonic to access your new wallet.
-          </Alert>
-        )}
-        <label className="authentication-label" htmlFor="mnemonic-submitted">
-          Enter the 12, 15, 24 or 27-word wallet mnemonic seed phrase
-        </label>
-        <input
-          type="text"
-          className="input fullwidth auth"
-          id="mnemonic-submitted"
-          name="mnemonic-submitted"
-          data-cy="MnemonicTextField"
-          placeholder="Enter your wallet mnemonic"
-          value={formData.mnemonicInputValue}
-          onInput={updateMnemonic}
-          onBlur={updateMnemonicValidationError}
-          autoComplete="off"
-          ref={(element) => {
-            this.mnemonicField = element
-          }}
-          onKeyDown={(e) => e.key === 'Enter' && this.goBtn.click()}
-        />
-        <div className="validation-row">
-          <button
-            className="button primary"
-            disabled={!formData.formIsValid}
-            onClick={async () =>
-              loadWallet({
-                cryptoProviderType: CryptoProviderType.WALLET_SECRET,
-                // TODO(ppershing): get rid of mnemonic sanitization in this component
-                walletSecretDef: await mnemonicToWalletSecretDef(sanitizedMnemonic),
-                shouldExportPubKeyBulk: true,
-              })
-            }
-            {...tooltip(
-              'Your input appears to be incorrect.\nCheck for the typos and try again.',
-              formData.mnemonicInputError
-            )}
-            onKeyDown={(e) => {
-              e.key === 'Enter' && (e.target as HTMLButtonElement).click()
-              if (e.key === 'Tab') {
-                this.mnemonicField.focus()
-                e.preventDefault()
-              }
-            }}
-            ref={(element) => {
-              this.goBtn = element
-            }}
-          >
-            Unlock
-          </button>
-          {formData.mnemonicInputError && (
-            <div className="validation-message error">
-              {getErrorMessage(formData.mnemonicInputError.code)}
-            </div>
+  const mnemonicField = useRef<HTMLInputElement>(null)
+  const goBtn = useRef<HTMLButtonElement>(null)
+
+  useEffect(() => {
+    if (!formData.mnemonicInputValue && !displayWelcome && !shouldShowDemoWalletWarningDialog) {
+      mnemonicField.current.focus()
+    }
+  }, [formData.mnemonicInputValue, displayWelcome, shouldShowDemoWalletWarningDialog])
+
+  // TODO: this reference always changes inside "useEffect" which causes infite loop.
```

Comment:
Please take a look at https://github.com/vacuumlabs/adalite/pull/938 it could solve this infinite loop issue,

Reply:
@xdzurman btw this PR was already merged, so you can rebase & simplify this

Reply:
@rizip1 Fixed